examples/proc_macro.rs

```diff
@@ -28,32 +28,30 @@ use jsonrpsee::{
 	proc_macros::rpc,
 	types::{async_trait, error::Error},
 	ws_client::WsClientBuilder,
-	ws_server::{RpcModule, SubscriptionSink, WsServerBuilder},
+	ws_server::{SubscriptionSink, WsServerBuilder},
 };
+use serde::de::DeserializeOwned;
+use serde::ser::Serialize;
 use std::net::SocketAddr;
 
 #[rpc(server, client, namespace = "state")]
-pub trait Rpc {
+pub trait Rpc<
+	Hash: DeserializeOwned + Serialize + Send + Sync + 'static,
```

Comment:
These trait bounds are required when both need to the client and server but we could add them behind the scenes instead.